lib/fluent/plugin/out_papertrail.rb

```diff
@@ -63,7 +63,8 @@ def send_to_papertrail(packet)
       @socket ||= create_socket(@papertrail_host, @papertrail_port)
 
       if @socket.nil?
-        log.warn "socket is nil -- failed to send: #{packet.assemble}"
+        log.error "socket is nil -- failed to send: #{packet.assemble}"
+        raise 'Unable to create socket with Papertrail'
```

Comment:
It's a best practice in Ruby to use a named exception class so that any upstream consumer can rescue that particular exception:

```ruby
class SocketFailureError < StandardError; end
...

raise SocketFailureError, "Unable to create socket connection to Papertrail"
```

Reply:
Thanks for the quick response and good call on multiple calls to packet.assemble. I'll reduce that and move to a named exception class.

Reply:
huzzah!